tests/phpunit/tests/theme/wpThemeJsonResolver.php

```diff
@@ -1070,4 +1070,79 @@ public function test_get_style_variations_returns_all_variations() {
 			$actual_settings
 		);
 	}
+
+	/**
+	 * @ticket 57545
+	 */
+	public function test_theme_shadow_presets_do_not_override_default_shadow_presets() {
+		switch_theme( 'block-theme' );
+
+		$theme_json_resolver = new WP_Theme_JSON_Resolver();
+		$theme_json          = $theme_json_resolver->get_merged_data();
+		$actual_settings     = $theme_json->get_settings()['shadow']['presets'];
+
+		$expected_settings = array(
+			'default'=> array(
+				array(
+					'name' => 'Natural',
+					'shadow' => '6px 6px 9px rgba(0, 0, 0, 0.2)',
+					'slug' => 'natural',
+				),
```

Comment:
It might be helpful to assert specifically that the `natural` present has not been overriden by the Theme version from `block-theme`'s `theme.json`.

The original bug related to the theme presets with the same slug overriding the core presets so it might be good to specifically assert to focus the test on this aspect.

Reply:
Maybe also a reference to the location of the core preset?

https://github.com/WordPress/gutenberg/blob/f85d5c9de9507a78958d732c3d3d28a664bbe038/lib/theme.json#L196-L200

Reply:
Maybe also include a reference to the location of the related core preset?

https://github.com/WordPress/wordpress-develop/blob/f25ca58803b886ab7e09f5ece878bb4e942a03b1/src/wp-includes/theme.json#L196-L200

Reply:
This value would never be overridden. What would happen instead is that `theme` array would have the `natural` preset. You may want to check by setting `shadow.defaultPresets:false` in core's theme.json and then execute `npm run test:php -- --filter test_theme_shadow_presets_do_not_override_default_shadow_presets`.

Note that I've also added a theme preset that is actually added (`test`) to make sure this is also working.

Reply:
Thank you for the additional clarification. I was understanding correctly, but perhaps not communicating that understanding very well.

When I say "overridden" I mean that the _bug_ was that the _Theme_ preset would take precedence over the Core preset.

Now with this PR the _Core_ preset takes precedence over the Theme preset.

The condition is that they must have a matching slug. This is why you've added a "Test" preset that only exists in the Theme to verify that this functionality still works.